dht.go

```diff
@@ -265,23 +265,38 @@ func (d *DHT) AddNode(addr string) {
 // Asks for more peers for a torrent.
 func (d *DHT) getPeers(infoHash InfoHash) {
 	closest := d.routingTable.lookupFiltered(infoHash)
-	for _, r := range closest {
-		// *remoteNode is nil if it got filtered.
-		if r != nil {
-			d.getPeersFrom(r, infoHash)
+	if len(closest) == 0 {
+		for _, s := range strings.Split(d.config.DHTRouters, ",") {
+			if s != "" {
+				r, e := d.routingTable.getOrCreateNode("", s, d.config.UDPProto)
+				if e == nil {
+					d.getPeersFrom(r, infoHash)
+				}
+			}
 		}
 	}
+	for _, r := range closest {
+		d.getPeersFrom(r, infoHash)
+	}
 }
 
 // Find a DHT node.
 func (d *DHT) findNode(id string) {
 	ih := InfoHash(id)
 	closest := d.routingTable.lookupFiltered(ih)
-	for _, r := range closest {
-		if r != nil {
-			d.findNodeFrom(r, id)
+	if len(closest) == 0 {
```

Comment:
Isn't this already guaranteed by the neighbor upkeep that is kicked off during startup?